tests/test_main.py

```
@@ -1199,6 +1712,30 @@ def _reset_global_state(self):
+                module._ble_future = None
+            if hasattr(module, "_ble_future_address"):
+                module._ble_future_address = None
+            if hasattr(module, "_ble_timeout_counts"):
+                module._ble_timeout_counts = {}
+            if hasattr(module, "_metadata_executor"):
+                executor = module._metadata_executor
+                if executor is not None:
+                    try:
+                        executor.shutdown(wait=False, cancel_futures=True)
+                    except Exception:
+                        pass
```

Ruff (0.14.13)

`try`-`except`-`pass` detected, consider logging the exception

(S110)